### PR TITLE
fix(shipping): CHECKOUT-5895 Keep billingSameAsShipping state when navigating through steps

### PR DESCRIPTION
**Pull request by RobinOng**
## What?
When navigating to billing/payment step and back to shipping step, the "My billing address is the same as my shipping address" checkbox always default to `true` 

## Why?
The state of the checkbox is not saved

## Testing / Proof

https://user-images.githubusercontent.com/22089936/124700192-e17cd680-df1e-11eb-818a-e58fbe3d8233.mov

@bigcommerce/checkout

### CODE REVIEW

File: src/app/checkout/Checkout.tsx [hunk 1]
```diff
@@ -100,6 +101,7 @@ class Checkout extends Component<CheckoutProps & WithCheckoutProps & WithLanguag
     stepTracker: StepTracker | undefined;
 
     state: CheckoutState = {
+        billingSameAsShipping: true,
```
Comment (animesh1987) [inline src/app/checkout/Checkout.tsx:104]: Is that the default behaviour we want to go to?

File: src/app/checkout/Checkout.tsx [hunk 2]
```diff
@@ -66,6 +66,7 @@ export interface CheckoutProps {
 
 export interface CheckoutState {
     activeStepType?: CheckoutStepType;
+    billingSameAsShipping: boolean;
```
Comment (animesh1987) [inline src/app/checkout/Checkout.tsx:69]: Should this have the `is` prefix as per other booleans in this interface?

File: src/app/checkout/Checkout.tsx [hunk 3]
```diff
@@ -343,6 +348,7 @@ class Checkout extends Component<CheckoutProps & WithCheckoutProps & WithLanguag
             >
                 <LazyContainer>
                     <Shipping
+                        billingSameAsShipping={ billingSameAsShipping }
```
Comment (animesh1987) [inline src/app/checkout/Checkout.tsx:351]: Same as above

File: src/app/checkout/Checkout.tsx (hunk 2 above)
Comment (RobinOng) [reply src/app/checkout/Checkout.tsx]: Good idea. I've added `is` prefix to it

File: src/app/checkout/Checkout.tsx (hunk 1 above)
Comment (RobinOng) [reply src/app/checkout/Checkout.tsx]: Yeah, it was the behaviour before the change as well. The purpose of this is to reduce friction in checkout

Comment (animesh1987) [reply src/app/checkout/Checkout.tsx]: Thanks for the explanation 👍

**Review by animesh1987 — APPROVED** [review]

**Comment by animesh1987** [comment]
The code looks good, there seem to be a few lint errors that circle is failing on.

File: src/app/shipping/BillingSameAsShippingField.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ const BillingSameAsShippingField: FunctionComponent<BillingSameAsShippingFieldPr
     return <CheckboxFormField
         id="sameAsBilling"
         labelContent={ labelContent }
-        name="billingSameAsShipping"
+        name="isBillingSameAsShipping"
```
Comment (davidchin) [inline src/app/shipping/BillingSameAsShippingField.tsx:20]: Is this change intentional? I believe it's going break any customisation that targets `[name=billingSameAsShipping]` so it's better not to change it. You can keep the name of the variable as `isBillingSameAsShipping` in other places, but not this one as it can affect the HTML output.

Comment (RobinOng) [reply src/app/shipping/BillingSameAsShippingField.tsx]: Good point. I've changed this back

**Review by davidchin — APPROVED** [review]
LGTM!
